crates/broker/src/order_picker.rs

```diff
@@ -867,10 +892,43 @@ where
 
                         let picker_clone = picker.clone();
                         let task_cancel_token = cancel_token.child_token();
+                        let order_id = order.id();
+
+                        // Track this task as active
+                        {
+                            let mut active_tasks = picker_clone.active_tasks.lock().await;
+                            active_tasks.insert(order_id.clone(), order.clone());
+
+                            // Log current active tasks
+                            let task_details = active_tasks
+                                .values()
+                                .map(ToString::to_string)
+                                .collect::<Vec<_>>()
+                                .join(", ");
+                            tracing::info!("Current pricing tasks: [{}]", task_details);
```

Comment:
This seems to verbose for an info log. Also, why not log outside of the loop for orders? This seems like it would just spam this log for each order added?

Reply:
Right, I did it such that whenever an order task starts or completes we log. I wanted it such that you can search an order id and see its full flow (when it was added/removed from the queue, start/stopped), and also see which other orders its competing with